pymilvus/orm/connections.py

```diff
@@ -199,9 +201,14 @@ def disconnect(self, alias: str):
         if not isinstance(alias, str):
             raise ConnectionConfigException(message=ExceptionsMessage.AliasType % type(alias))
 
-        if alias in self._connected_alias:
-            self._connected_alias.pop(alias).close()
-
+        with self._con_lock:
+            if alias in self._connected_alias:
+                gh = self._connected_alias.pop(alias)
+                self._connection_references[id(gh)] -= 1
+                if self._connection_references[id(gh)] <= 0:
+                    gh.close()
```

Comment:
It seems that `close` is not necessary to be called, see also https://github.com/grpc/grpc/issues/19235.

Reply:
It seems like its not worth reference tracking then. Instead of tracking we just reuse if available and keep them open until the program terminates. This could result in OOM in a long running program though. What do you think? Reuse channels but risk OOM or use new channels but have clean close? @xiaofan-luan @longjiquan

Reply:
Figured out my error, I called wait_for_channel_ready on an already made channel. Guessing waiting for ready only works when unready.